src/hotspot/cpu/aarch64/compressedKlass_aarch64.cpp

```diff
@@ -130,4 +130,14 @@ void CompressedKlassPointers::initialize(address addr, size_t len) {
   _base = (end <= (address)unscaled_max) ? nullptr : addr;
 
   _range = end - _base;
+
+  DEBUG_ONLY(assert_is_valid_encoding(addr, len, _base, _shift);)
+}
+
+bool CompressedKlassPointers::pd_is_valid_encoding(address addr, size_t len, address base, int shift) {
+  size_t range = (addr == nullptr) /* Not specified */ ? len : (addr + len - base);
+  return (base == nullptr /* Zero-based encoding */ ||
+          (1ULL << MacroAssembler::eor_compatible_klass_encoding((uint64_t)base) >= range) ||
+          MacroAssembler::movk_compatible_klass_encoding((uint64_t)base)) &&
+         (shift == -1 /* Not specified */ || shift == 0);
```

Comment:
This duplicates the assert we have in MacroAssembler; why do this twice, in two places?